### PR TITLE
Ignore EngineVersion minor updates when `autoMinorVersionUpgrade` is enabled

### PR DESCRIPTION
**Pull request by joaquinservetto**
Fixes [Issue](https://github.com/aws-controllers-k8s/community/issues/2009)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/resource/db_cluster/custom_update.go [hunk 1]
```diff
@@ -561,7 +562,12 @@ func (rm *resourceManager) newCustomUpdateRequestPayload(
 		res.SetEnableIAMDatabaseAuthentication(*desired.ko.Spec.EnableIAMDatabaseAuthentication)
 	}
 	if desired.ko.Spec.EngineVersion != nil && delta.DifferentAt("Spec.EngineVersion") {
-		res.SetEngineVersion(*desired.ko.Spec.EngineVersion)
+		r := regexp.MustCompile(`[0-9]*$`)
```
Comment (a-hilaly) [inline pkg/resource/db_cluster/custom_update.go:565]: Can we delcare this regex as a total level file variable? just avoid compiling the regex at every reconciliation loop

File: pkg/resource/db_cluster/custom_update.go [hunk 2]
```diff
@@ -561,7 +562,12 @@ func (rm *resourceManager) newCustomUpdateRequestPayload(
...
+		desiredMajorEngineVersion := r.ReplaceAllString(*desired.ko.Spec.EngineVersion, "${1}")
```
Comment (a-hilaly) [inline pkg/resource/db_cluster/custom_update.go:566]: nil, we need to handle the case of nil `*desired.ko.Spec.EngineVersion`

File: pkg/resource/db_cluster/custom_update.go [hunk 3]
```diff
@@ -561,7 +562,12 @@ func (rm *resourceManager) newCustomUpdateRequestPayload(
 		res.SetEnableIAMDatabaseAuthentication(*desired.ko.Spec.EnableIAMDatabaseAuthentication)
 	}
 	if desired.ko.Spec.EngineVersion != nil && delta.DifferentAt("Spec.EngineVersion") {
```
Comment (a-hilaly) [inline pkg/resource/db_cluster/custom_update.go:564]: Can we move this logic into a little function called `requireEngineVersionUpdate` or something similar? that was we could add unit tests for it :)

**Review by a-hilaly — COMMENTED** [review]
This is great start, muchas gracias @joaquinservetto ! i left few recommendations below
/ok-to-test

File: pkg/resource/db_cluster/custom_update.go (hunk 2 above)
Comment (joaquinservetto) [reply pkg/resource/db_cluster/custom_update.go]: Thanks a lot for you review @a-hilaly. If `*desired.ko.Spec.EngineVersion` is nil it'll not perform any update, isn't it the expected behavior?

**Review by a-hilaly — APPROVED** [review]
Thank you @joaquinservetto !
/lgtm

**Pull request merged**